tikv-client/src/main/java/com/pingcap/tikv/types/Converter.java

```diff
@@ -176,4 +180,60 @@ public static BigDecimal convertToBigDecimal(Object val) {
       throw new TypeException("can not cast non Number type to Double");
     }
   }
+
+  public static String convertDurationToStr(long nanos, int decimal) {
+    int sign = 1, hours, minutes, seconds, frac;
+    if (nanos < 0) {
+      nanos = -nanos;
+      sign = -1;
+    }
+    hours = (int) (nanos / HOUR);
+    nanos -= hours * HOUR;
+    minutes = (int) (nanos / MINUTE);
+    nanos -= minutes * MINUTE;
+    seconds = (int) (nanos / SECOND);
+    nanos -= seconds * SECOND;
+    frac = (int) (nanos / MICROSECOND);
+    StringBuilder sb = new StringBuilder();
+    if (sign < 0) {
+      sb.append('-');
+    }
+    sb.append(String.format("%02d:%02d:%02d", hours, minutes, seconds));
+    if (decimal > 0) {
+      sb.append('.');
+      sb.append(String.format("%06d", frac), 0, decimal);
+    }
+    return sb.toString();
+  }
+
+  public static long convertStrToDuration(String value) {
+    requireNonNull(value, "value cannot be null");
+    // value should be in form of 12:59:59.000 or 12:59:59
+    // length expect to be 3.
+    String[] splitBySemiColon = value.split(":");
+    if (splitBySemiColon.length != 3)
+      throw new IllegalArgumentException(
+          String.format("%s is not a valid time type in mysql", value));
```

Comment:
Please provide correct format string to user in error message. It can be a const string and also in above cases use the same.

Reply:
DONE